### ISSUE TITLE
Hang while running test suite for `scalatestplus-scalacheck` (possible regression)

### ISSUE THREAD

**Issue by griggt** [typelevel/scalacheck#909]
Running the test suite for [scalatestplus-scalacheck](https://github.com/scalatest/scalatestplus-scalacheck) appears to hang indefinitely (while shrinking?) when built against the latest commit on the scalacheck main branch (7749c38313f9deb79368d72f592f7f1e9383620d).

The same test suite passes within about 5 seconds if built against the previous merge commit (dc93d7954264e4135be39d699c6130f3160d3af6), so it seems related to #838.

This issue was noticed while updating projects in the Scala 3 community build (https://github.com/lampepfl/dotty/pull/15728#issuecomment-1196246968) however it is also reproducible on Scala 2.13.

### Reproduction Steps

1. `publishLocal` scalacheck @ 7749c38313f9deb79368d72f592f7f1e9383620d
2. clone https://github.com/scalatest/scalatestplus-scalacheck/commit/32e7dc67a9490147e6a04be9125c3ef0940d6819
3. update `build.sbt` of scalatestplus-scalacheck to use the locally published scalacheck:
```diff
-    "org.scalacheck" %%% "scalacheck" % "1.16.0",
+    "org.scalacheck" %%% "scalacheck" % "1.17-7749c38-SNAPSHOT",
```
4. run `sbt scalatestPlusScalaCheckJVM/test`

**Comment by armanbilge** [comment]
Thanks for the report. To ease reproduction the mentioned snapshot is actually published.
https://s01.oss.sonatype.org/content/repositories/snapshots/org/scalacheck/scalacheck_2.13/1.17-7749c38-SNAPSHOT/

**Comment by rossabaker** [comment]
I've been meaning to do a release, but we'll hold for this fix.  Thanks for catching it!

**Comment by rossabaker** [comment]
I haven't tested with scalatestplus-scalacheck, but it should be fixed in 1.17-8689ae7-SNAPSHOT.
